ins-app/app/src/main/java/ins/developers/indoornavigationsystem/MainActivity.java

```diff
@@ -0,0 +1,71 @@
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MainActivity extends AppCompatActivity {
+    final String TAG = this.getClass().getSimpleName();
+    TextView responseText;
+    Button getRequestButton;
+    EditText employeeSelector;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        responseText = (TextView) findViewById(R.id.result);
+        getRequestButton = (Button) findViewById(R.id.get_button);
+        employeeSelector = (EditText) findViewById(R.id.node_setter);
+
+        getRequestButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                // Instantiate the RequestQueue.
+                RequestQueue queue = Volley.newRequestQueue(getApplicationContext());
+                String selectedEmployee = employeeSelector.getText().toString();
+                String employeeId = selectedEmployee.length() != 0 ? selectedEmployee : "1"; // Default employee id
+                String url ="http://10.0.2.2:8050/get_device_pos/" + employeeId; //TO-DO: Change to get_employee_pos
+
+                // Request a string response from the provided URL.
+                JsonObjectRequest stringRequest = new JsonObjectRequest(Request.Method.GET, url,
+                        new Response.Listener<JSONObject>() {
+                            @Override
+                            public void onResponse(JSONObject response) {
+                                try {
+                                    String xVal = response.getString("pos_x");
+                                    String yVal = response.getString("pos_y");
+                                    String zVal = response.getString("pos_z");
+                                    responseText.setText("X:" + xVal + " Y:" + yVal + " Z:" + zVal);
+                                } catch (JSONException e) {
+                                    responseText.setText("Error parsing server response!");
+                                    e.printStackTrace();
+                                }
+                            }
+                        }, new Response.ErrorListener() {
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+                        Log.e(TAG, error.toString());
+                        responseText.setText("Error, employee not found!");
+                    }
+                });
+                // Add the request to the RequestQueue.
+                queue.add(stringRequest);
+            }
+        });
```

Comment:
Interesting to see this approach of making http request.

Reply:
It seems to be Google's [suggested way](https://developer.android.com/training/volley/request.html) of doing things.

I am not very fond of it to be honest, I had trouble figuring out a way to test all those inline functions/callbacks. By looking around I found some examples on how to do that but it wasn't completely straight forward.